tdp/core/models/deployment_log.py

```diff
@@ -241,6 +242,72 @@ def from_operations(
                 i += 1
         return deployment_log
 
+    @staticmethod
+    def from_operations_hosts_vars(
+        collections: Collections,
+        operation_host_vars_names: list[tuple[str, Optional[str], Optional[list[str]]]],
+    ) -> "DeploymentLog":
+        """Generate a deployment plan from a list of operations, hosts and extra vars.
+
+        Args:
+            collections: Collections to retrieve the operations from.
+            operation_host_vars_names: List of operations, hosts and extra vars to perform.
+
+        Raises:
+            UnsupportedOperationError: If an operation is a noop.
+            ValueError: If an operation is not found in the collections.
+        """
+        (operation_names, host_names, var_names) = [
+            list(item) for item in zip(*operation_host_vars_names)
+        ]
+        collections.check_operations_exist(operation_names)
+
+        #   -  hosts_list : list of hosts, sorted and without duplicates ; if an operation is executed on eveyr host, the result is None (all the hosts)
+        #   -  extra_vars_list : list of extra vars (which is list of lists), sorted and without duplicates
+        hosts_list = list(sorted(set(host_names))) if None not in host_names else None
+        extra_vars_list = (
+            list(
+                sorted(
+                    set(
+                        list(
+                            itertools.chain(
+                                *[item for item in var_names if item is not None]
+                            )
+                        )
+                    )
+                )
+            )
+            if var_names
+            else None
+        )
+
+        deployment_log = DeploymentLog(
+            targets=operation_names,
+            hosts=hosts_list,
+            extra_vars=extra_vars_list if extra_vars_list else None,
+            deployment_type=DeploymentTypeEnum.OPERATIONS,
+            state=DeploymentStateEnum.PLANNED,
+        )
+        i = 1
+        for (operation_name, host_name, var_names) in operation_host_vars_names:
```

Comment:
You can use the `enumerate` method to avoid having to use `i`.

Reply:
Done.